File: javascript/prime-factors/prime-factors.js
function PrimeFactors() {

}

PrimeFactors.prototype.for = function(input) {

  var divisor = 2;
  var result = [];
  var current = input;

  while(current > 1)
  {
    if((current % divisor) == 0){
      result.push(divisor);
      current = current / divisor;
    }else{
      divisor++;
    }
  }

  return result;
}

module.exports = new PrimeFactors()

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors

    def self.for(input)
        res = []
        divisor = 2
        current = input

        while current > 1
            if current % divisor == 0
                res.push(divisor)
                current = current / divisor
            else
                divisor += 1
            end
        end

        res
    end

end